### DISCUSSION TITLE
rest api and webapp at the sametime

### DISCUSSION THREAD

**Discussion by wanghaisheng** [fastapi/fastapi#4773] [Questions] [answered]
app = FastAPI()
apiapp = FastAPI()

@apiapp.get("/e1/", response_class=ORJSONResponse)
async def e1(url:str):

@apiapp.get("/e2/", response_class=ORJSONResponse)
async def e2(url:str):

def index() -> None:

home = asgi_app(index)
app.mount("/", home)
app.mount("/e1", apiapp)
app.mount("/e2", apiapp)

e1  e2  still 404

**Comment by jgould22** [comment, marked as answer]
You appear to be mounting your apiapp on a path, so your endpoints will be `/e1/e1/` or `/e2/e2/` if you only want to have it at `/e1` then change 
```python 
@apiapp.get("/e1/", response_class=ORJSONResponse)
```
to 
```python 
@apiapp.get("/", response_class=ORJSONResponse)
```
and the same for your `e2`

**Reply by wanghaisheng** [reply]
i end up with change 
```

app.mount("/", home)
app.mount("/e1", apiapp)
app.mount("/e2", apiapp)
```
to
```
apiapp.mount("/", home)
```
since there is router app.get 
I dont need mount api endpoint ,just mount a asgi  webview app (flask or pywebio etc)
am i right?
